diff --git a/serverModules/firebaseDB.js b/serverModules/firebaseDB.js
index e69de29..bbd08eb 100644
--- a/serverModules/firebaseDB.js
+++ b/serverModules/firebaseDB.js
@@ -22,7 +22,11 @@ const createAppInFirestore = async (appData) => {
   const privateId = generatePrivateId(); // Implement this function based on your requirements.

   const newAppData = {
-    privateId: privateId,
+    privateId,
+    secret: generatePrivateSecret(), // Added a new field 'secret' for editing rights
+  };
+
+function generatePrivateSecret() {
+  // Returns a randomly generated secret for editing rights
+  return Math.random().toString(36).substring(2, 15) + Math.random().toString(36).substring(2, 15);
+}

   const docRef = await db.collection('Apps').add(newAppData);
   return { id: docRef.id, privateId: privateId };

